cmd/soroban-rpc/internal/db/migration.go

```diff
@@ -179,30 +162,58 @@ func (g *guardedMigration) Commit(ctx context.Context) error {
 	if g.alreadyMigrated {
 		return nil
 	}
-	err := setMetaBool(ctx, g.db, g.guardMetaKey, true)
-	if err != nil {
-		return errors.Join(err, g.Rollback(ctx))
-	}
-	return g.db.Commit()
+	return setMetaBool(ctx, g.db, g.guardMetaKey, true)
 }
 
-func (g *guardedMigration) Rollback(_ context.Context) error {
-	return g.db.Rollback()
+func GetMigrationLedgerRange(ctx context.Context, db *DB, retentionWindow uint32) (*LedgerSeqRange, error) {
+	firstLedgerToMigrate := firstLedger
+	latestLedger, err := NewLedgerEntryReader(db).GetLatestLedgerSequence(ctx)
+	if err != nil && !errors.Is(err, ErrEmptyDB) {
+		return nil, fmt.Errorf("failed to get latest ledger sequence: %w", err)
+	}
+	if latestLedger > retentionWindow {
```

Comment:
Double-checking: this behavior is sane if `errors.Is(err, ErrEmptyDb)`, right? i.e. `latestLedger = 0` in that case and everything else seems like it checks out? But I think that'd make `firstLedgerToMigrate > latestLedger`, no?

Reply:
hmm I think it should err out in that case too cc @aditya1702 would know it better. can this happen?